### PR TITLE
Add note about new binaries to changelog

### LINKED ISSUE #25686: Update changelog and docs for docker proxy changes
It looks like the 1.12.0 changelog doesn't have the docker proxy changes made in https://github.com/docker/docker/pull/23312. I don't think the docs have been updated either.

### PR DESCRIPTION
**Pull request by thaJeztah**
fixes https://github.com/docker/docker/issues/25686

ping @tiborvass PTAL; not sure if we should have the "IMPORTANT" section added (although it won't hurt)

### CODE REVIEW

**Comment by justincormack** [comment]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,9 @@ To resolve this:
 After making those changes, run `sudo systemctl daemon-reload`, and `sudo
 systemctl restart docker` to reload changes and (re)start the docker daemon.
 
+**IMPORTANT**: With Docker 1.12, a Linux docker installation now has two
+additional binaries; `dockerd`, and `docker-proxy`. If you have scripts for
```
Comment (tiborvass) [inline CHANGELOG.md]: how is `dockerd` specific to 1.12 ?

Comment (thaJeztah) [inline CHANGELOG.md]: `dockerd` wasn't in 1.11 yet (yup, I also thought it was); https://github.com/docker/docker/pull/20639

**Comment by vdemeester** [comment]
LGTM 🐸

**Pull request merged**